WL-4061 URL for a non-electronic citation can be deleted now.

ouit0408 opened the pull request:
Empty  url is a valid url for non-electronic citation, which gets stored as null in DB therefore have 
added null and empty checks in the template.

nicholaswilson100 commented on citations-tool/tool/src/java/org/sakaiproject/citation/tool/CitationHelperAction.java:
```diff
@@ -5321,10 +5321,14 @@ protected ContentResource createTemporaryResource(ResourceToolActionPipe pipe)
 		{
 			throw new MalformedURLException();
 		}
+		if(url.equals(""))
```
You've removed the .trim() from above ...Was that redundant?

nicholaswilson100 replied:
What's the difference between a null and empty string for a url here?  Does null mean something has gone wrong and empty string that they've deleted a url?

ouit0408 replied:
For a non-electronic citation an empty string is a vaild url but not null.

ouit0408 replied:
by mistake removed the trim method, will add back.

ouit0408 committed:
WL-4061 URL for a non-electronic citation can be deleted now.

Empty  url is a valid url for non-electronic citation therefore have added
null and empty checks in the template.
